qiskit/ignis/calibrations/fitters.py

```diff
@@ -0,0 +1,379 @@
+        """
+
+        iq_list_mean, iq_list_var = self._build_iq_list()
+
+        self._ydata = {}
+        for _, serieslbl in enumerate(self._series):
+            self._ydata[serieslbl] = []
+            for qind, _ in enumerate(self._qubits):
+                self._ydata[serieslbl].append({'mean': [], 'std': []})
+                mean_list = self._ydata[serieslbl][-1]['mean']
+                var_list = self._ydata[serieslbl][-1]['std']
+                for circ, _ in enumerate(self._xdata):
+                    circname = self._circuit_names[circ] + serieslbl
+                    mean_list.append(iq_list_mean[circname][qind])
+                    var_list.append(iq_list_var[circname][qind])
+                    # problem for the fitter if one of the std points is
+                    # exactly zero
+                    if var_list[-1] == 0:
+                        var_list[-1] = 1e-4
+
+                # project the data onto a line
+                # mean over all the experiment
+                q_exp_mean = np.mean(mean_list)
+                mean_list -= q_exp_mean
+                real_ext = np.mean(np.abs(np.real(mean_list)))
+                imag_ext = np.mean(np.abs(np.imag(mean_list)))
+                crot = real_ext/(real_ext**2+imag_ext**2)**0.5
+                srot = imag_ext/(real_ext**2+imag_ext**2)**0.5
+                mean_list = crot*np.real(mean_list)+srot*np.imag(mean_list)
+                self._ydata[serieslbl][-1]['mean'] = mean_list
+
+    def plot(self, qind, series='0', ax=None, show_plot=True):
+        """
+        Plot calibration data.
+
+        Args:
+            qind: qubit index to plot
+            ax: plot axes
+            show_plot: call plt.show()
+
+        Returns:
+            The axes object
+        """
+
+        from matplotlib import pyplot as plt
+
+        if ax is None:
+            plt.figure()
+            ax = plt.gca()
+
+        ax.errorbar(self._xdata, self._ydata[series][qind]['mean'],
+                    self._ydata[series][qind]['std'],
+                    marker='.', markersize=9,
+                    c='b', linestyle='')
+        ax.plot(self._xdata, self._fit_fun(self._xdata,
+                                           *self._params[series][qind]),
+                c='r', linestyle='--',
+                label='Q%d' % (self._qubits[qind]))
+
+        ax.tick_params(axis='x', labelsize=14, labelrotation=70)
+        ax.tick_params(axis='y', labelsize=14)
+        ax.set_xlabel('X', fontsize=16)
+        ax.set_ylabel('IQ Signal', fontsize=16)
+        ax.set_title(self._description + ' for qubit ' +
+                     str(self._qubits[qind]), fontsize=18)
+        ax.legend(fontsize=12)
+        ax.grid(True)
+        if show_plot:
+            plt.show()
+
+        return ax
+
+
+class RabiFitter(BaseCalibrationFitter):
+    """
+    Rabi Experiment fitter
+    """
+
+    def __init__(self, backend_result, xdata,
+                 qubits,
```

Comment:
All the data is in a single result, so it seems like a lot of unnecessary object creation and rerunning the same code

Reply:
The code already runs most of the code twice in the loops, also the API is more complicated for having all of the qubits, and it's not necessarily straightforward on how to split them apart. It will also become confusing when we start to calibrate two-qubit gates. In which case what do qubits mean? Would it be a list of two-tuples of qubits?

Reply:
There's a two qubit fitter already, I forget what I did